assembly/collections/persistentMap.ts

```diff
@@ -206,29 +220,36 @@ export class PersistentMap<K, V> {
    * @param key - Key of the element.
    * @param value - The new value of the element.
    */
-  set(key: K, value: V): void {
-    // assert map size wont overflow
-    assert(this._size < Usize.MAX_VALUE);
+  set(key: K, value: V): Result<usize> {
+    // check for map size wont overflow
+    if (this._size >= Usize.MAX_VALUE) {
+      return new Result(this.size(), 'map size overflow');
+    }
     if (isString<V>()) {
       this._increaseSize(key);
       // @ts-ignore
-      Storage.set(this._key(key), value);
+      Storage.set(this._key(key), stringToBytes(value));
     } else if (isInteger<V>()) {
       this._increaseSize(key);
       // @ts-ignore
-      Storage.set(this._key(key), value.toString());
+      Storage.set(this._key(key), i64ToBytes(value as i64));
     } else if (isFloat<V>()) {
       this._increaseSize(key);
       // @ts-ignore
-      Storage.set(this._key(key), value.toString());
+      Storage.set(this._key(key), f64ToBytes(value as f64));
     } else if (isBoolean<V>()) {
       this._increaseSize(key);
       // @ts-ignore
-      Storage.set(this._key(key), value.toString());
+      Storage.set(this._key(key), boolToByte(value as boolean));
+    } else if (isArrayLike<V>()) {
+      this._increaseSize(key);
+      // @ts-ignore
+      Storage.set(this._key(key), unwrapStaticArray(value));
     } else {
       this._increaseSize(key);
       // @ts-ignore
       Storage.set(this._key(key), value.toString());
     }
+    return new Result(this.size(), null);
```

Comment:
I removed some @ts-ignore with `as` and moved _increaseSize before `if`

```ts

  /**
   * @example
   * ```ts
   * let map = new PersistentMap<string, string>("m")
   *
   * map.set("hello", "world")
   * ```
   *
   * Sets the new value for the given key.
   * @param key - Key of the element.
   * @param value - The new value of the element.
   */
  set(key: K, value: V): Result<usize> {
    // check for map size wont overflow
    if (this._size >= Usize.MAX_VALUE) {
      return new Result(this.size(), 'map size overflow');
    }

    this._increaseSize(key);

    if (isString<V>()) {
      Storage.set(this._key(key), stringToBytes(value as string));
    } else if (isInteger<V>()) {
      Storage.set(this._key(key), i64ToBytes(value as i64));
    } else if (isFloat<V>()) {
      Storage.set(this._key(key), f64ToBytes(value as f64));
    } else if (isBoolean<V>()) {
      Storage.set(this._key(key), boolToByte(value as boolean));
    } else if (isArrayLike<V>()) {
      Storage.set(this._key(key), unwrapStaticArray(value as Uint8Array));
    } else {
      // @ts-ignore
      Storage.set(this._key(key), value.toString());
    }

    return new Result(this.size());
  }
```

Reply:
done, thanks